admin/admin-default-term.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Manages filters and actions related to default terms.
+ *
+ * @since 3.0
+ */
+class PLL_Admin_Default_Term {
+
+	/**
+	 * A reference to the PLL_Model instance.
+	 *
+	 * @since 2.8
+	 *
+	 * @var PLL_Model
+	 */
+	protected $model;
+
+	/**
+	 * Preferred language to assign to new contents.
+	 *
+	 * @var PLL_Language
+	 */
+	protected $pref_lang;
+
+	/**
+	 * Reference to Polylang options array
+	 *
+	 * @var array $options
+	 */
+	protected $options;
+
+	/**
+	 * @param object $polylang
+	 */
+	public function __construct( &$polylang ) {
+		$this->model = &$polylang->model;
+		$this->pref_lang = &$polylang->pref_lang;
+		$this->options = &$polylang->options;
+	}
+
+	/**
+	 * Setups filters and actions needed.
+	 *
+	 * @return void
+	 */
+	public function add_hooks() {
+		$taxonomies = get_taxonomies();
+		foreach ( $taxonomies as $taxonomy ) {
+			if ( 'category' === $taxonomy ) {
```

Comment:
This is the only line where I should read `category`. Everywhere else (function names, filters, actions, options), you need to remove the reference to the category to mak this class fully generic.